### PR TITLE
browser split part 2

### PR DESCRIPTION
**Pull request by RonnyPfannschmidt**
* introduces a factory concept
* steamline configuration overrides and browser
* distinct wharf factory
* defer all browser creation to the factory

### CODE REVIEW

**Comment by RonnyPfannschmidt** [comment]
it needs a extra pair of eyes on the wharf part

**Comment by dajoRH** [comment]
Lint report for commit 25d35ade000e8d75f8fa118b905b11d542c28d19:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

File: utils/browser.py [hunk 1]
```diff
@@ -24,31 +24,204 @@
...
+                self.wharf.checkout()
+                return super(WharfFactory, self).create()
+            except urllib2.URLError as ex:
+                # connection to selenum was refused for unknown reasons
+                logger.error('URLError connecting to selenium; recycling container. URLError:')
+                write_line('URLError caused container recycle, see log for details', red=True)
+                logger.exception(ex)
+                self.wharf.checkin()
+
+        def renew(self):
```
Comment (psav) [inline utils/browser.py]: Should this be indented?

Comment (RonnyPfannschmidt) [inline utils/browser.py]: very good find!

File: utils/browser.py [hunk 2]
```diff
@@ -24,31 +24,204 @@
...
+        browser.file_detector = UselessFileDetector()
+        browser.maximize_window()
+        browser.get(store.base_url)
+        return browser
+
+
+class WharfFactory(BrowserFactory):
+    def __init__(self, webdriver_class, browser_kwargs, wharf):
+        super(WharfFactory, self).__init__(webdriver_class, browser_kwargs)
+        self.whaf = wharf
```
Comment (psav) [inline utils/browser.py]: ```self.wharf```

File: utils/browser.py [hunk 3]
```diff
@@ -24,31 +24,204 @@
...
+            browser_kwargs['firefox_profile'] = self._firefox_profile
+
+    @cached_property
+    def _firefox_profile(self):
+        return _load_firefox_profile()
+
+    def renew(self):
+        pass
+
+    def processed_bowser_args(self):
```
Comment (mfalesni) [inline utils/browser.py]: `bowser`

**Comment by dajoRH** [comment]
Lint report for commit 14892150443b158140eee08891610ce3fd188726:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
You have multiple commits in the PR. When you want your PR to be merged, please squash them. Do not squash them when working on the pull request as this allows to see the progress!

*CFME QE Bot*

**Comment by psav** [comment]
Let's squash and we can merge

**Pull request merged**